src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -215,21 +216,43 @@ public unsafe void Start()
                 _receivePause = new ManualResetEventSlim(true);
                 _reachability.StatusChanged += ReachabilityChanged;
                 _reachability.Start();
+
                 // STEP 1: Create the TcpClient, which is responsible for negotiating
                 // the socket connection between here and the server
-                try {
-                    // ReSharper disable once UseObjectOrCollectionInitializer
-                    _client = new TcpClient(AddressFamily.InterNetworkV6);
-                } catch (Exception e) {
-                    DidClose(e);
-                    return;
-                }
 
-                try {
-                    _client.Client.DualMode = true;
-                } catch (ArgumentException) {
-                    WriteLog.To.Sync.I(Tag, "IPv4/IPv6 dual mode not supported on this device, falling back to IPv4");
-                    _client = new TcpClient(AddressFamily.InterNetwork);
+                if (!String.IsNullOrEmpty(_options.NetworkInterface)) {
+                    try {
+                        var localAddress = GetLocalNIAddress(_options.NetworkInterface);
+                        if (localAddress == null) {
+                            WriteLog.To.Sync.I(Tag, $"Unknown Network Interface {_options.NetworkInterface}.");
+                            DidClose(new CouchbaseNetworkException(C4NetworkErrorCode.UnknownHost));
```

Comment:
I don't think this error is appropriate.  Did you find it in another platform implementation?

Reply:
Pasin said to match URLEndpointListener Config Network Interface invalid error. And base on the test https://github.com/couchbase/couchbase-lite-net/blob/master/src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs#L615
new CouchbaseNetworkException(C4NetworkErrorCode.UnknownHost) is the exception with error. Please suggest a better one if you think it makes more sense.

Reply:
If the other one uses it then it's probably fine.  It's not likely to happen that often anyway.